### PR TITLE
add gruntfile.js for minify task, minified source, and test runner against minified source

### PR DESCRIPTION
**Pull request by sjmiles**
Demonstrates #65, allows testing against `grunt-contrib-uglify`

### CODE REVIEW

File: gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    'wrappers/Element.js',
+    'wrappers/HTMLElement.js',
+    'wrappers/HTMLContentElement.js',
+    'wrappers/HTMLShadowElement.js',
+    'wrappers/HTMLTemplateElement.js',
+    'wrappers/HTMLUnknownElement.js',
+    'wrappers/generic.js',
+    'wrappers/ShadowRoot.js',
+    'ShadowRenderer.js',
+    'wrappers/Document.js',
```
Comment (arv) [inline gruntfile.js:24]: This is not up to date... it should have `'wrappers/Window.js'` as well

File: test/index.min.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+})</script>
+<script src="test.js"></script>
+<script src="paralleltrees.js"></script>
+<script src="rerender.js"></script>
+<script src="reprojection.js"></script>
+<script src="custom-element.js"></script>
+<script src="HTMLShadowElement.js"></script>
+<script src="HTMLContentElement.js"></script>
+<script src="HTMLTemplateElement.js"></script>
+<script src="Document.js"></script>
```
Comment (arv) [inline test/index.min.html]: Missing `Window.js`

**Comment by arv** [comment]
Is there a better way for us to not duplicate all this code?

One thing we can do is to at least move the js in the html file into a separate js file.

**Comment by sjmiles** [comment]
For the other polyfills we made the <polyfill>.js file conditionally load
<polyfill>.debug.js or <polyfill>.min.js based on flags. In general that
lets us have single HTML files.

I don't know if you want to go that far.

**Comment by arv** [comment]
We can have js file that has all the known files in it... like a manifest.

File: gruntfile.js (hunk 1 above)
Comment (sjmiles) [inline gruntfile.js:24]: fixed

**Comment by sjmiles** [comment]
Take a look at that. I factored 95% of the source into `test.main.js` which I load from `test*.html`.

File: test/test.main.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    assert.equal(a[i], b[i], msg);
+  }
+}
+
+mocha.setup({
+  ui: 'tdd',
+  globals: ['console', 'getInterface']
+})
+
+modules = [
```
Comment (arv) [inline test/test.main.js]: missing var?

'use strict' would have told you so.

**Comment by arv** [comment]
LGTM

Please squash before merging.

**Comment by sjmiles** [comment]
superseded by squashed pull #74

**Pull request closed without merging**